_episodes/07-raster-calculations.md

```diff
@@ -3,303 +3,367 @@ title: "Raster Calculations in Python"
+![](../fig/E09-01-PONE-NDVI.jpg)
+
+Source: Wu C-D, McNeely E, Cedeño-Laurent JG, Pan W-C, Adamkiewicz G, Dominici F, et al. (2014) Linking Student Performance in Massachusetts Elementary Schools with the “Greenness” of School Surroundings Using Remote Sensing. PLoS ONE 9(10): e108548. https://doi.org/10.1371/journal.pone.0108548
+{: .text-center}
 
 > ## More Resources
-> * Check out more on LiDAR CHM, DTM and DSM in this NEON Data Skills overview tutorial:
-> [What is a CHM, DSM and DTM? About Gridded, Raster LiDAR Data](https://www.neonscience.org/chm-dsm-dtm-gridded-lidar-data).
+> Check out more on NDVI in the NASA Earth Observatory portal:
+> [Measuring Vegetation](https://earthobservatory.nasa.gov/features/MeasuringVegetation/measuring_vegetation_2.php).
 {: .callout}
 
-### Load the Data
-For this episode, we will use the DTM and DSM from the
-NEON Harvard Forest Field site and San Joaquin Experimental
-Range, which we already have loaded from previous episodes. 
-Let's load them again with `open_rasterio` using the argument `masked=True`.
+### Load and crop the Data
+For this episode, we will use one of the Sentinel-2 scenes that we have already employed in the previous episodes. Let's
+load the results of our initial imagery search using `pystac`:
 
-```python
+~~~
+import pystac
+items = pystac.ItemCollection.from_file("search.json")
+~~~
+{: .language-python}
+
+We then select the second item, and extract the URIs of the red and NIR bands ("B04" and "B8A", respectively):
+
+~~~
+red_uri = items[1].assets["B04"].href
+nir_uri = items[1].assets["B8A"].href
+~~~
+{: .language-python}
+
+Let's load the rasters with `open_rasterio` using the argument `masked=True`.
+
+~~~
 import rioxarray
+red = rioxarray.open_rasterio(red_uri, masked=True)
+nir = rioxarray.open_rasterio(nir_uri, masked=True)
+~~~
+{: .language-python}
+
+Let's also restrict our analysis to the same crop field area defined in the previous episode by clipping the rasters
+using a bounding box:
 
-surface_HARV = rioxarray.open_rasterio("data/NEON-DS-Airborne-Remote-Sensing/HARV/DSM/HARV_dsmCrop.tif", masked=True)
-terrain_HARV_UTM18 = rioxarray.open_rasterio("data/NEON-DS-Airborne-Remote-Sensing/HARV/DTM/HARV_dtmCrop_UTM18.tif", masked=True)
-```
+~~~
+bbox = (629000, 5804000, 639000, 5814000)
+red_clip = red.rio.clip_box(*bbox)
+nir_clip = nir.rio.clip_box(*bbox)
+~~~
+{: .language-python}
+
+We can now plot the two rasters. Using `robust=True` color values are stretched between the 2nd and 98th percentiles of
+the data, which results in clearer distinctions between high and low reflectances:
+
+~~~
+red_clip.plot(robust=True)
+~~~
+{: .language-python}
+
+![](../fig/E09-02-red-band.png)
+
+~~~
+nir_clip.plot(robust=True)
+~~~
+{: .language-python}
+
+![](../fig/E09-03-NIR-band.png)
+
+It is immediately evident how crop fields (rectangular shapes in the central part of the two figures) appear as dark and
+bright spots in the red-visible and NIR wavelengths, respectively.
```

Comment:
Could it be relevant here to mention also the seasonality of the crop fields? I assume that in this case the figure is plotted at a time when the crops are already grown? For a time period in the off season, you might see dark spots even in the crop fields.

Reply:
Nice idea, added a sentence to highlight the concept!